src/main.rs

```diff
@@ -187,13 +195,18 @@ impl TryFrom<Args> for Config {
         println!("Using identity {}", node_keys.pk());
 
         let force_proxy = args.proxy.is_some();
-        let proxy_addr = args.proxy.unwrap_or(Localhost::localhost()).into();
+        let proxy_addr = args
+            .proxy
+            .flatten()
+            .unwrap_or(Localhost::localhost())
+            .into();
```

Comment:
here we detect do we need to force proxy for non-onion addresses or not - but we provide proxy info always